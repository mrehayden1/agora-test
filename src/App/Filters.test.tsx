import { shallow } from 'enzyme';
import React from 'react';

import Filter from './Filters';

test('renders matching snapshot', () => {
  const component = shallow(<Filter />);
  expect(component).toMatchSnapshot();
});

test('invokes onSortChange correctly when sort is changed', () => {
  const spy = jest.fn();
  const component = shallow(<Filter onSortChange={spy} />);
  component
    .find('#sort-input').simulate('change', { target: { value: 'hello' }});
  expect(spy).toBeCalledWith('hello');
});

test('invokes onSearchChange correctly when sort is changed', () => {
  const spy = jest.fn();
  const component = shallow(<Filter onSearch={spy} />);
  component
    .find('#search-input').simulate('input', { target: { value: 'search change' }});
  expect(spy).toBeCalledWith('search change');
});
