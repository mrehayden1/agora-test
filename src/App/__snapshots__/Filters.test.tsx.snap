// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders matching snapshot 1`] = `
<form
  className="filters"
>
  <h2>
    Filters
  </h2>
  <div
    className="control"
  >
    <label
      htmlFor="sort-input"
    >
      Sort by: 
    </label>
    <select
      id="sort-input"
      name="sort"
      onChange={[Function]}
    >
      <option
        value="newest"
      >
        Newest
      </option>
      <option
        value="oldest"
      >
        Oldest
      </option>
      <option
        value="name"
      >
        Name (A-Z)
      </option>
      <option
        value="name-desc"
      >
        Name (Z-A)
      </option>
    </select>
  </div>
  <div
    className="control"
  >
    <label
      htmlFor="search-input"
    >
      Search mission name: 
    </label>
    <input
      id="search-input"
      name="search"
      onInput={[Function]}
    />
  </div>
</form>
`;
